buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt

```diff
@@ -135,6 +139,29 @@ open class ShadedJarPlugin : Plugin<Project> {
         }
     }
 
+    private
+    fun Project.defaultShadedJarLocation() =
+        layout.buildDirectory.file("shaded-jar/${base.archivesBaseName}-shaded-${rootProject.extra["baseVersion"]}.jar")
+
+    private
+    fun Project.addInstallShadedJarTask(shadedJarTask: TaskProvider<ShadedJar>) {
+        val installPathProperty = "${project.name}ShadedJarInstallPath"
+        val taskName = "install${project.name.capitalize()}ShadedJar"
+        fun targetFile() = findProperty(installPathProperty)?.let { File(findProperty(installPathProperty) as String) }
+        if (true == targetFile()?.isAbsolute) {
+            tasks.register<Copy>(taskName) {
+                dependsOn(shadedJarTask)
+                from(defaultShadedJarLocation())
```

Comment:
Why don't we use the property of the `ShadedJar` task directly? You can `map` the `TaskProvider` to obtain the property.

Reply:
Good to know, thanks!